### ISSUE TITLE
Stream Join problem

### ISSUE THREAD

**Issue by nesmaalshazly** [confluentinc/ksql#7935]
Dears

 I have use case which can be described as follow
1- Dump that is generated each day at 4 am
2- online stream that is run from 12 am for 24 hours

we use dump as lookup , any content that exist in online stream and also in the dump will have special offer
but we face problem as our proposed solution is limited
1- we created stream that join between lookup dump stream and online stream for 24

but we face problem as there is gap because dump is not ready before 4 am so the join find nothing in those 4 hours

and if we changed window period for more time we will lose each day refreshment data

any help?

**Comment by mjsax** [comment]
This is tricky... You need to understand that a joins have temporal semantics. For a stream-table join, it implies that a stream record joins to the table "version" that is valid according to the event-time of the stream record. Thus, if you table updates are timestamps at 4am, all stream-records from 12am-4am happen _before_ the table update and are not eligible "see" those updates.

And even if you can change the timestamp of the table updates to 12am, the issue is that ksqlDB does not support a GRACE PRIOD for stream-table joins yet (we are already working on this though)...

Bottom line might be, that you would need to work around the current ksqlDB limitations to change your upstream ingestion  to "wait" until after all table updates got published, before you publish the stream updates... Not sure if this is possible in your end-to-end setup.
